web/quickfeed_service.go

```diff
@@ -361,6 +361,15 @@ func (s *QuickFeedService) DeleteGroup(ctx context.Context, in *connect.Request[
 	return &connect.Response[qf.Void]{}, nil
 }
 
+func (s *QuickFeedService) GetSubmission(ctx context.Context, in *connect.Request[qf.SubmissionReviewersRequest]) (*connect.Response[qf.Submission], error) {
+	submission, err := s.db.GetSubmission(&qf.Submission{ID: in.Msg.GetSubmissionID()})
```

Comment:
So server-side does not need courseID?

Reply:
We need to pass the course ID to pass the `teacher` check in `access_control.go`:

```go
case teacher:
    if method == "GetUserByCourse" {
        if err := claims.IsCourseTeacher(tm.Database(), request.Any().(*qf.CourseUserRequest)); err != nil {
            return nil, connect.NewError(connect.CodePermissionDenied, fmt.Errorf("AccessControl(%s): %w", method, err))
        }
	return next(ctx, request)
    }
    if claims.HasCourseStatus(req, qf.Enrollment_TEACHER) {
	return next(ctx, request)
    }
```
However, a teacher for course X, with claims for course X, will pass this check. I have no checks implemented yet that also check if the supplied submission ID actually belongs to course X. Currently requesting a submission related to course Y, with claims only for X, will still return the requested submission.

This could likely be avoided by looking up and checking the `CourseID` of the assignment related to the requested submission.